polaris-react/src/components/Pagination/Pagination.scss

```diff
@@ -24,4 +24,43 @@
       }
     }
   }
+
+  &.table {
+    button {
+      #{$se23} & {
```

Comment:
Actually... this seems to introduce a regression, because the existing styles with `se23` at the top of the file will have higher specificity and these styles need to override them. So, the `table` class will need to have `se23` scoped styles while the base Pagination styles have `se23` scoped styles.